File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    result = []
    
    for  i in range(len(score)):
        
        if len(answer) < k:
            answer.append(score[i])
            result.append(min(answer))
        else:
            if score[i] > min(answer):
                del answer[answer.index(min(answer))]
                answer.append(score[i])
                result.append(min(answer))
            else:
                result.append(min(answer))
            
    
    
    return result

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        ArrayList<Integer> result = new ArrayList<>();
        ArrayList<Integer> honor = new ArrayList<>();
        
        
        for(int i : score){
            
            honor.add(i);
            Collections.sort(honor);
            
            if(honor.size() > k){
                honor.remove(0);
            }
            
            result.add(honor.get(0));
        }
        
        int[] answer = result.stream().mapToInt(Integer::intValue).toArray();        
        return answer;
    }
}